### PR TITLE
refactor: grid item details toggle example improvements

### PR DESCRIPTION
**Pull request by sissbruecker**
Closes https://github.com/vaadin/docs/issues/3986

### CODE REVIEW

File: frontend/demo/component/grid/react/grid-item-details-toggle.tsx [hunk 1]
```diff
@@ -25,6 +27,24 @@ function Example() {
     });
   }, []);
 
+  const toggleDetailsRenderer = useCallback(({ item: person }: { item: Person }) => {
```
Comment (sissbruecker) [inline frontend/demo/component/grid/react/grid-item-details-toggle.tsx:30]: Redeclaring renderer functions every render cycle has issues at the moment (https://github.com/vaadin/react-components/issues/283), so extracted this.

File: frontend/demo/component/grid/grid-item-details-toggle.ts [hunk 2]
```diff
@@ -80,28 +83,35 @@ export class Example extends LitElement {
...
-        <vaadin-grid-column path="displayName" header="Name"></vaadin-grid-column>
-        <vaadin-grid-column path="profession"></vaadin-grid-column>
         <vaadin-grid-column
-          ${columnBodyRenderer<Person>(
-            (person) => html`
+          width="80px"
+          flex-grow="0"
+          frozen
+          ${columnBodyRenderer<Person>((person) => {
+            const isExpanded = this.detailsOpenedItems.includes(person);
```
Comment (vursen) [inline frontend/demo/component/grid/grid-item-details-toggle.ts:91]:
```suggestion
          ${columnBodyRenderer<Person>((person, { detailsOpened }) => {
```

File: src/main/java/com/vaadin/demo/component/grid/GridItemDetailsToggle.java [hunk 3]
```diff
@@ -42,7 +45,19 @@ public GridItemDetailsToggle() {
     private static Renderer<Person> createToggleDetailsRenderer(
             Grid<Person> grid) {
         return LitRenderer.<Person> of(
-                "<vaadin-button theme=\"tertiary\" @click=\"${handleClick}\">Toggle details</vaadin-button>")
+                """
+                <vaadin-button
+                  theme="tertiary icon"
+                  aria-label="Toggle details"
+                  aria-expanded="${item.detailsVisible ? 'true' : 'false'}"
```
Comment (vursen) [inline src/main/java/com/vaadin/demo/component/grid/GridItemDetailsToggle.java:52]:
```suggestion
                  aria-expanded="${model.detailsOpened ? 'true' : 'false'}"
```

File: frontend/demo/component/grid/grid-item-details-toggle.ts [hunk 4]
```diff
@@ -80,28 +83,35 @@ export class Example extends LitElement {
...
-                  this.detailsOpenedItems = isOpened
+                  this.detailsOpenedItems = isExpanded
                     ? this.detailsOpenedItems.filter((p) => p !== person)
                     : [...this.detailsOpenedItems, person];
                 }}"
               >
-                Toggle details
+                ${isExpanded
+                  ? html` <vaadin-icon icon="lumo:angle-down"></vaadin-icon>`
+                  : html` <vaadin-icon icon="lumo:angle-right"></vaadin-icon>`}
```
Comment (web-padawan) [inline frontend/demo/component/grid/grid-item-details-toggle.ts:105]: I think we could replace this with a single icon e.g. like this:

```html
<vaadin-icon
  .icon="${isExpanded ? 'lumo:angle-down' : 'lumo:angle-right'}"
></vaadin-icon>
```

**Review by web-padawan — APPROVED** [review]

File: frontend/demo/component/grid/grid-item-details-toggle.ts (hunk 2 above)
Comment (sissbruecker) [reply frontend/demo/component/grid/grid-item-details-toggle.ts]: Thanks, done. Did not change this for the React example, as it currently requires typing props manually which would make this more confusing IMO: https://github.com/vaadin/react-components/issues/284

File: src/main/java/com/vaadin/demo/component/grid/GridItemDetailsToggle.java (hunk 3 above)
Comment (sissbruecker) [reply src/main/java/com/vaadin/demo/component/grid/GridItemDetailsToggle.java]: Done

File: frontend/demo/component/grid/grid-item-details-toggle.ts (hunk 4 above)
Comment (sissbruecker) [reply frontend/demo/component/grid/grid-item-details-toggle.ts]: Makes sense, done

**Pull request merged**